### PR TITLE
[TextFields] Adding Color Themer.

### PR DESCRIPTION
**Pull request by mohammadcazig**
Implemented the full Color Themer to theme our textfields and inputControllers.
Pivotal: https://www.pivotaltracker.com/story/show/156392456

Screenshot of themed textfields:
screen shot 2018-04-05 at 3 31 00 pm

### CODE REVIEW

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h [hunk 1]
```diff
@@ -16,9 +16,11 @@
 
 #import <Foundation/Foundation.h>
 
-@protocol MDCColorScheme;
+@protocol MDCColorScheming;
```
Comment (jverkoey) [inline components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: This should be an import of the `MaterialColorScheme.h` header. Generally-speaking we avoid forward declaring symbols from components that aren't in the current target (e.g. we wouldn't forward declare UIView or UITableViewDelegate).

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h [hunk 2]
```diff
@@ -16,9 +16,11 @@
...
 @protocol MDCTextInputController;
 
+@class MDCTextField;
```
Comment (jverkoey) [inline components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: Similarly here - import `MaterialTextFields.h`.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h [hunk 3]
```diff
@@ -33,7 +35,7 @@
  @param colorScheme The color scheme to apply.
  @param textInputController A MDCTextInputController instance to apply a color scheme.
  */
-+ (void)applyColorScheme:(nonnull id<MDCColorScheme>)colorScheme
++ (void)applyColorScheme:(nonnull id<MDCColorScheming>)colorScheme
```
Comment (jverkoey) [inline components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: As discussed offline, we unfortunately can't change these APIs without it being an instant breaking change for all downstream clients.

**Review by jverkoey — CHANGES_REQUESTED** [review]
API and import feedback.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h [hunk 4]
```diff
@@ -17,13 +17,54 @@
...
+ Performs introspection to determine if class passed conforms to
+ MDCTextInputControllerFloatingPlaceholder to set properties like
+ .floatingPlaceholderNormalColorDefault.
+
+ @param colorScheme The color scheme to apply.
+ @param textInputControllerClass A Class that conforms to MDCTextInputController (at least.)
+ */
++ (void)applySemanticColorScheme:(nonnull id<MDCColorScheming>)colorScheme
+toAllTextInputControllersOfClass:(nonnull Class<MDCTextInputController>)textInputControllerClass
+NS_SWIFT_NAME(apply(_:toAllControllersOfClass:));
```
Comment (jverkoey) [inline components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: stylenit: four-space indentation

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h [hunk 5]
```diff
@@ -50,4 +91,5 @@
 + (void)applyColorScheme:(nonnull id<MDCColorScheme>)colorScheme
```
Comment (jverkoey) [inline components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h:94]: Add comments to the old APIs mapping to the new APIs that should be used instead.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h (hunk 1 above)
Comment (romoore) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: When did we start avoiding forward declarations in headers? The last I heard was there was no recommendation either way?

Comment (jverkoey) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: We're not avoiding all forward declarations, just ones for APIs that aren't part of our target.

Comment (romoore) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: Ok. Is this captured somewhere, I've been adding forward declarations as much as possible recently.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h (hunk 5 above)
Comment (mohammadcazig) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h:94]: Done.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h (hunk 1 above)
Comment (mohammadcazig) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: Done.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h (hunk 2 above)
Comment (mohammadcazig) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: Done.

File: components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h (hunk 3 above)
Comment (mohammadcazig) [reply components/TextFields/src/ColorThemer/MDCTextFieldColorThemer.h]: Done.

**Review by jverkoey — APPROVED** [review]
Looks good. Please add a screenshot of a text field themed with the material defaults.

**Comment by mohammadcazig** [comment]
I had some merging issue that effected this PR as well. The diff looks good so I am switching the label to cla:yes.

**Pull request merged**